tests/libnet/ipaddress.go

```diff
@@ -0,0 +1,29 @@
+package libnet
+
+import (
+	v1 "k8s.io/api/core/v1"
+	netutils "k8s.io/utils/net"
+)
+
+func GetPodIpByFamily(pod *v1.Pod, family v1.IPFamily) string {
+	var ips []string
+	for _, ip := range pod.Status.PodIPs {
+		ips = append(ips, ip.IP)
+	}
+	return getIp(ips, family)
+}
+
+func getIp(ips []string, family v1.IPFamily) string {
+	for _, ip := range ips {
+		if netutils.IsIPv6String(ip) {
+			if family == v1.IPv6Protocol {
```

Comment:
what about adding a function that return the family(ip) and then you can just
if family == getFamily(ip)
  return ip
instead all the 3 ifs

more over, using a singleton to determine the position, assuming positioning isnt dynamic might can be considered,
but not a must

Reply:
I don't understand what do you mean by "using a singleton to determine the position". Can you please explain?

Reply:
have a bool that tells you if its the first time then func run,
in case its the first time, 
calculate POS_IPv4 and POS_IPv6 which can be 0, 1, or -1 if NA
and from now only if its not the first time just use the POS_X

Reply:
Well, I believe this method traversing over an array with *2* items won't cause us any performance issue. So let's keep it readable.

Reply:
Regarding your first suggestion. That's how the code will look like.

`func getIp(ips []string, family v1.IPFamily) string {
	for _, ip := range ips {
		if family == getFamily(ip) {
			return ip
		}
	}
	return ""
}

func getFamily(ip string) v1.IPFamily {
	if netutils.IsIPv6String(ip) {
		return v1.IPv6Protocol
	} 
	return v1.IPv4Protocol
}`

I'm not sure it is better than the original one. That's a pretty simple method, let's not over engineer it.

Reply:
i am totally against over eng, 
in this case i dont find it over eng, but nicer, and more modern, remind python shortcuts
you can also do this i think:

```
	for _, ip := range ips; family == getFamily(ip) {
            return ip
	}
```
in case you want to reconsider please

Reply:
I don't think that's a valid syntax. I will use `getFamily`. The "singleton" approach it too much for me:)